src/nvim/os/process.c

```diff
@@ -89,21 +89,17 @@ bool os_proc_tree_kill(int pid, int sig)
 bool os_proc_tree_kill(int pid, int sig)
 {
   assert(sig == SIGTERM || sig == SIGKILL);
-  int pgid = getpgid(pid);
-  if (pgid > 0) {  // Ignore error. Never kill self (pid=0).
-    if (pgid == pid) {
-      ILOG("sending %s to process group: -%d",
-           sig == SIGTERM ? "SIGTERM" : "SIGKILL", pgid);
-      int rv = uv_kill(-pgid, sig);
-      return rv == 0;
-    } else {
-      // Should never happen, because process_spawn() did setsid() in the child.
-      ELOG("pgid %d != pid %d", pgid, pid);
-    }
+  if (pid != 0) {  // Never kill self (pid=0).
+    ILOG("sending %s to PIDs %d and %d",
+         sig == SIGTERM ? "SIGTERM" : "SIGKILL", pid, -pid);
+
+    // TODO(vim): have an option to only kill the process, not the group?
+    int rv1 = uv_kill(-pid, sig);
+    int rv2 = uv_kill(pid, sig);
+    return (rv1 == 0) && (rv2 == 0);
```

Comment:
Not sure about the return value - maybe `rv2` is false, if the first killing is fast?

Reply:
why is `uv_kill(pid)`, shouldn't it die with the group? Isn't this a race?

Reply:
meh, i see https://github.com/vim/vim/issues/3656#issuecomment-445459692 .

anyways, getting rid of `getpgid` is good. There's no need for it anyway, libuv understands "negative PID" on all platforms.

Reply:
The return value is not used really..

Reply:
My comment above isn't related to the return value. Return value is given "just in case" a caller might care to know whether the operation succeeded.

Reply:
Will change it to `(rv1 == 0) || (rv2 == 0)`.

Reply:
did you see the edited comment suggesting swapping the order of calls: https://github.com/neovim/neovim/pull/10660#discussion_r308696198

Reply:
(had not seen your edited comment)
Heh.. I've amended it before based on your other comment (before editing).. (https://github.com/neovim/neovim/issues/9704#issuecomment-471331759).

Reply:
> That avoids a race where the pid dies with its process group, then OS creates an unrelated process with the same PID, then we kill that PID.

Could still happen, but only with a new pgid then turning up?

Reply:
Trying only -pgid now - which is really better if it works.
As for the test failure with this in Vim: it might be due to Vim doing things different in this regard after all.  Let's see if it causes some of our tests to fail also, and then we might want to cover it explicitly.